src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -117,12 +122,25 @@ public Workspace Workspace
     /// <summary>
     /// A list of all the ids for all the projects contained by the solution.
     /// </summary>
-    public IReadOnlyList<ProjectId> ProjectIds => this.SolutionState.ProjectIds;
+    public IReadOnlyList<ProjectId> ProjectIds
```

Comment:
>public IReadOnlyList<ProjectId> ProjectIds [](http://example.com/codeflow?start=4&length=42)

I consider this a bit of a wart on this change, but it's a public api so my hands are tied. I am considering changing the remaining callers in Roslyn to instead use ProjectStates to reduce the likelihood of needing to allocate this (this PR changed all usages in the workspaces folder).